server/models.py

```diff
@@ -942,22 +946,65 @@ class GradingTask(Model):
     @hybrid_property
     def is_complete(self):
         return self.score_id is not None
+        # return self.kind in [s.tag for s in self.backup.scores]
+
+    @hybrid_property
+    def total_tasks(self):
+        tasks = (GradingTask.query
+                            .filter_by(grader_id=self.grader_id,
+                                       assignment_id=self.assignment_id)
+                            .count())
+        return tasks
+
+    @hybrid_property
+    def completed(self):
+        completed = (GradingTask.query
+                                .filter_by(grader_id=self.grader_id,
+                                           assignment_id=self.assignment_id)
+                                .filter(GradingTask.score_id)
+                                .count())
+        return completed
 
     @hybrid_property
     def remaining(self):
         ungraded = (GradingTask.query
-                              .filter_by(grader_id=self.grader_id,
-                                         assignment_id=self.assignment_id,
-                                         score_id=None)
-                              .count())
+                               .filter_by(grader_id=self.grader_id,
+                                          assignment_id=self.assignment_id,
+                                          score_id=None)
+                               .count())
         return ungraded
-        # return self.kind in [s.tag for s in self.backup.scores]
 
     def get_next_task(self):
         ungraded = (GradingTask.query
-                              .filter_by(grader_id=self.grader_id,
-                                         assignment_id=self.assignment_id,
-                                         score_id=None)
-                              .order_by(GradingTask.created.asc())
-                              .first())
+                               .filter_by(grader_id=self.grader_id,
+                                          assignment_id=self.assignment_id,
+                                          score_id=None)
+                               .order_by(GradingTask.created.asc())
+                               .first())
         return ungraded
+
+    @classmethod
+    def get_staff_tasks(cls, assignment_id):
+        """ Return list of namedtuple objects that represent queues.
+            Only uses 1 SQL Query.
+            Attributes:
+                - grader: User, assigned grader
+                - completed: int, completed tasks
+                - remaining: int, ungraded tasks
+        """
+        tasks = (db.session.query(cls,
```

Comment:
It would be ideal if this only selected the grader_id and the two counts. It might also resolve the error from ONLY_GROUP_BY